File: GeneratedEntityReaderClasses/Components/LaserEmitterComponent.cs
namespace GeneratedNoitaClasses;
public class LaserEmitterComponent : NoitaComponentBase,  iNoitaType<LaserEmitterComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public ConfigLaser laser { get; set; } = new();
	public NoitaBool is_emitting { get; set; } = new();
	public NoitaInt emit_until_frame { get; set; } = new();
	public NoitaFloat laser_angle_add_rad { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		laser.Read(s);
		is_emitting.Read(s);
		emit_until_frame.Read(s);
		laser_angle_add_rad.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		laser.Write(s);
		is_emitting.Write(s);
		emit_until_frame.Write(s);
		laser_angle_add_rad.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/LaserEmitterComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from GeneratedClasses.ConfigLaser import ConfigLaser
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class LaserEmitterComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    laser:ConfigLaser = field(default_factory=ConfigLaser)
    is_emitting:NoitaBool = field(default_factory=NoitaBool)
    emit_until_frame:NoitaInt = field(default_factory=NoitaInt)
    laser_angle_add_rad:NoitaFloat = field(default_factory=NoitaFloat)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.laser.Read(s)
        self.is_emitting.Read(s)
        self.emit_until_frame.Read(s)
        self.laser_angle_add_rad.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.laser.Write(s)
        self.is_emitting.Write(s)
        self.emit_until_frame.Write(s)
        self.laser_angle_add_rad.Write(s)

